plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewReaderUtil.java

```diff
@@ -164,7 +164,8 @@ public static boolean isTrinoMaterializedView(String tableType, Map<String, Stri
     public static boolean canDecodeView(Table table)
     {
         // we can decode Hive or Presto view
-        return table.getTableType().equals(VIRTUAL_VIEW.name());
+        return table.getTableType().equals(VIRTUAL_VIEW.name()) &&
+                !isTrinoMaterializedView(table.getTableType(), table.getParameters());
```

Comment:
This was proposed in https://github.com/trinodb/trino/pull/15221 but back then we couldn't test it. Now we can thanks to the test added by Grant.
`canDecodeView` should be removed (https://github.com/trinodb/trino/pull/17276#discussion_r1180038413), leaving this to a follow-up (https://github.com/trinodb/trino/pull/17276).